### ISSUE TITLE
Slashes in db names?

### ISSUE THREAD

**Issue by remcovanvugt** [redgeoff/spiegel#179]
Hi,

It seems spiegel is not picking up changes in databases with names containing (escaped) slashes.
eg the spiegel replicator doc contains:

"source": "http://mysuser@mycouchhost.com/db%2Fname%2Fname2"
"dest": "http://myuser@mycouchreplhost.com/db%2Fname%2Fname2"

Initially the db replicates, but when I add a doc or delete a doc, the change is not replicated.

In the same setup, this works for databases with names without slashes.

In /_global_changes the db names are displayed in the JSON without escaping the slashes. How is this handled by the update listener? The source database is Couchdb 2.3.0, same problem on Couchdb 2.2.0.

Thanks.

**Comment by remcovanvugt** [comment]
I did some further research into this. I found out the _global_changes has db names like this:

update:db/name/name

_toDBname in update_listeners.js strips off update: and uses the rest as-is as db name. The Spiegel replicator doc has source and target specified as URL-safe names. Otherwise replication would fail I would assume.

Therefore, in the clean_or_locked_replicators_by_db_name view, the keys are URL-safe names as well. That doesn't match with with array keys used internally in spiegel so the db never gets marked dirty and therefore doesn't replicate. I added a very sysadmin-style "fix" by URL encoding the db names. It works, but I'm sure there will be a more sophisticated fix. I'll PR just for the sake of helping out pointing out my issue. Thanks.
